lib/myxql/connection.ex

```diff
@@ -438,7 +450,7 @@ defmodule MyXQL.Connection do
         {:ok, result, state}
 
       other ->
-        result(other, statement, state)
+        transaction_error(other, statement, state)
```

Comment:
```suggestion
        with {:error, _exception, state} <- result(other, statement, state) do
          {:error, state}
        end
```

So we can re-use all the old functionality

Reply:
can't get used to `with` operator since its introduction :) But sure, that is better. Fixed now.